apis/externalsecrets/v1/secretstore_gcpsm_types.go

```diff
@@ -60,3 +62,71 @@ type GCPSMProvider struct {
 	// Location optionally defines a location for a secret
 	Location string `json:"location,omitempty"`
 }
+
+// GCPWorkloadIdentityFederation holds the configurations required for generating federated access tokens.
+type GCPWorkloadIdentityFederation struct {
+	// credConfig holds the configmap containing the GCP external account credential configuration in JSON format and the key name containing the json data.
+	// +kubebuilder:validation:Required
+	CredConfig *ConfigMapReference `json:"credConfig,omitempty"`
+
+	// audience is the Secure Token Service (STS) audience which contains the resource name for the workload identity pool and the provider identifier in that pool.
+	// If specified, Audience found in the external account credential config will be overridden with the configured value.
+	// +kubebuilder:validation:Optional
+	Audience string `json:"audience,omitempty"`
+
+	// ServiceAccountRef is the reference to the kubernetes ServiceAccount to be used for obtaining the tokens,
+	// when Kubernetes is configured as provider in workload identity pool.
+	// +kubebuilder:validation:Optional
+	ServiceAccountRef esmeta.ServiceAccountSelector `json:"serviceAccountRef"`
+
+	// ExternalTokenEndpoint is the endpoint explicitly set up to provide tokens, which will be matched against the
+	// credential_source.url in the provided credConfig. This field is merely to double-check the external token source
+	// URL is having the expected value.
+	// +kubebuilder:validation:Optional
+	ExternalTokenEndpoint string `json:"externalTokenEndpoint,omitempty"`
```

Comment:
When credentials config as a URL configured in `credential_source`, which could be self created service to issue tokens, I thought to make use of `ExternalTokenEndpoint` to assert user knows about the configuration. I am not sure if it's really required though :)